aws_lambda_powertools/utilities/feature_flags/feature_flags.py

```diff
@@ -132,50 +148,48 @@ def get_feature_toggle(
                the defined feature
             3. feature exists and a rule matches -> rule_default_value of rule is returned
         """
-        if rules_context is None:
-            rules_context = {}
+        if context is None:
+            context = {}
 
         try:
-            toggles_dict: Dict[str, Any] = self.get_configuration()
+            features = self.get_configuration()
         except ConfigurationError:
-            logger.error("unable to get feature toggles JSON, returning provided value_if_missing value")
-            return value_if_missing
+            logger.debug("Unable to get feature toggles JSON, returning provided default value")
+            return default
 
-        feature: Dict[str, Dict] = toggles_dict.get(schema.FEATURES_KEY, {}).get(feature_name, None)
+        feature = features.get(name)
         if feature is None:
-            logger.warning(
-                f"feature does not appear in configuration, using provided value_if_missing, "
-                f"feature_name={feature_name}, value_if_missing={value_if_missing}"
+            logger.debug(
+                f"feature does not appear in configuration, using provided default, name={name}, default={default}"
             )
-            return value_if_missing
+            return default
 
         rules_list = feature.get(schema.RULES_KEY)
         feature_default_value = feature.get(schema.FEATURE_DEFAULT_VAL_KEY)
         if not rules_list:
-            # not rules but has a value
+            # no rules but value
             logger.debug(
-                f"no rules found, returning feature default value, feature_name={feature_name}, "
+                f"no rules found, returning feature default value, name={name}, "
                 f"default_value={feature_default_value}"
             )
             return bool(feature_default_value)
+
         # look for first rule match
-        logger.debug(
-            f"looking for rule match,  feature_name={feature_name}, feature_default_value={feature_default_value}"
-        )
+        logger.debug(f"looking for rule match,  name={name}, feature_default_value={feature_default_value}")
         return self._handle_rules(
-            feature_name=feature_name,
-            rules_context=rules_context,
+            feature_name=name,
```

Comment:
@heitorlessa seems like some times it is `name` and other times it is `feature_name`.

Reply:
Yep I'm still making a list of variables name to go after.

I hope I can get to the point where these can be abstracted by a higher level class so we reduce the amount of variable propagation

Reply:
Or better yet, handle exceptions higher up

Reply:
addressed.